components/event-publisher-proxy/pkg/handler/handler_test.go

```diff
@@ -225,3 +256,499 @@ func waitForHandlerToStart(t *testing.T, healthEndpoint string) {
+		if tc.wantedResult != got {
+			t.Errorf("incorrect result with inputURI: %s, wanted: %v, got: %v", tc.inputURI, tc.wantedResult, got)
+		}
+	}
+}
+
+// validateErrorResponse validates Error Response
+func validateErrorResponse(t *testing.T, resp http.Response, tcWantResponse *legacyapi.PublishEventResponses) {
+	legacyResponse := legacyapi.PublishEventResponses{}
+	legacyError := legacyapi.Error{}
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("failed to read response body: %v", err)
+	}
+	t.Logf("response body : %s", string(bodyBytes))
+	if err = json.Unmarshal(bodyBytes, &legacyError); err != nil {
+		t.Fatalf("failed to unmarshal response body: %v", err)
+	}
+	legacyResponse.Error = &legacyError
+	if !reflect.DeepEqual(tcWantResponse.Error, legacyResponse.Error) {
+		t.Fatalf("Invalid error, want: %v, got: %v", tcWantResponse.Error, legacyResponse.Error)
+	}
+}
+
+// validateOkResponse validates Ok Response
+func validateOkResponse(t *testing.T, resp http.Response, tcWantResponse *legacyapi.PublishEventResponses) {
+	legacyOkResponse := legacyapi.PublishResponse{}
+	legacyResponse := legacyapi.PublishEventResponses{}
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("failed to read response body: %v", err)
+	}
+	t.Logf("response body : %s", string(bodyBytes))
+	if err = json.Unmarshal(bodyBytes, &legacyOkResponse); err != nil {
+		t.Fatalf("failed to unmarshal response body: %v", err)
+	}
+	legacyResponse.Ok = &legacyOkResponse
+	if err = resp.Body.Close(); err != nil {
+		t.Fatalf("failed to close body: %v", err)
+	}
+
+	if tcWantResponse.Ok.EventID != "" && tcWantResponse.Ok.EventID != legacyResponse.Ok.EventID {
+		t.Errorf("invalid event-id, want: %v, got: %v", tcWantResponse.Ok.EventID, legacyResponse.Ok.EventID)
+	}
+
+	if tcWantResponse.Ok.EventID == "" && !isValidEventID(legacyResponse.Ok.EventID) {
+		t.Errorf("should match regex: [%s] Not a valid event-id: %v ", legacy.AllowedEventIDChars, legacyResponse.Ok.EventID)
+	}
+	if tcWantResponse.Ok.Reason != legacyResponse.Ok.Reason {
+		t.Errorf("invalid reason, want: %v, got: %v", tcWantResponse.Ok.Reason, legacyResponse.Ok.Reason)
+	}
+	if tcWantResponse.Ok.Status != legacyResponse.Ok.Status {
+		t.Errorf("invalid status, want: %v, got: %v", tcWantResponse.Ok.Status, legacyResponse.Ok.Status)
+	}
+}
+
+// generatePort generates a random 5 digit port
+func generatePort() (int, error) {
+	max := 4
+	// Add 4 as prefix to make it 5 digits but less than 65535
+	add4AsPrefix := "4"
+	b := make([]byte, max)
+	n, err := io.ReadAtLeast(rand.Reader, b, max)
+	if n != max {
+		return 0, err
+	}
+	if err != nil {
+		return 0, err
+	}
+	for i := 0; i < len(b); i++ {
+		b[i] = table[int(b[i])%len(table)]
+	}
+
+	num, err := strconv.Atoi(fmt.Sprintf("%s%s", add4AsPrefix, string(b)))
+	if err != nil {
+		return 0, err
+	}
+
+	return num, nil
+}
+
+var table = [...]byte{'1', '2', '3', '4', '5', '6', '7', '8', '9'}
```

Comment:
Does it make sense to use `math/rand` instead of `crypto/rand`? The implementation would be much simpler.

Reply:
I strongly believe crypto/rand has edges over math/rand and would like to leave it as is. This is even [Go-playground](https://play.golang.org/p/_lYUjO2RkuN) proof.